### ISSUE TITLE
Switching server dB name dynamically

### ISSUE THREAD

**Issue by mdonogma** [Mimetis/Dotmim.Sync#218]
Depending on client syncing I need to switch the server dB.

Example: client a and b sync with dB 1
Client c and d sync with server db2.

Think of multitenancy scenario where each tenant has its own server dB.

**Comment by Mimetis** [comment]
Each client should have a different  pointing to a different address

``` cs
// Client A & B
var wco1 = new WebClientOrchestrator("http://Server01/api/Sync");

// Client C & D
var wco2 = new WebClientOrchestrator("http://Server02/api/Sync");

```
Then create two ASP.NET Core Web Api, one for each server db

**Comment by mdonogma** [comment]
Is there another way, there could be 1000s of tenants

Sent from my iPhone

On 16 Feb 2020, at 22:58, Sébastien Pertus <notifications@github.com> wrote:

﻿

Each client should have a different pointing to a different address

// Client A & B
var wco1 = new WebClientOrchestrator("http://Server01/api/Sync");

// Client C & D
var wco2 = new WebClientOrchestrator("http://Server02/api/Sync");


Then create two ASP.NET Core Web Api, one for each server db

**Comment by Mimetis** [comment]
You will have to explain better what is your architecture, to allow me to help you

**Comment by mdonogma** [comment]
10000 users , 100 each having own tenant dB so many tenant dbs
Single web front end
Single sql server

Sent from my iPhone

On 17 Feb 2020, at 05:24, Sébastien Pertus <notifications@github.com> wrote:

﻿

You will have to explain better what is your architecture, to allow me to help you

**Comment by Mimetis** [comment]
I see.
Let me think about it, it seems to be possible, but needs some new features to support it on the web server side

**Comment by mdonogma** [comment]
Will be fantastic as I foresee many developers running into this scenario

Sent from my iPhone

On 17 Feb 2020, at 22:05, Sébastien Pertus <notifications@github.com> wrote:

﻿

I see.
Let me think about it, it seems to be possible, but needs some new features to support it on the web server side

**Comment by Mimetis** [comment]
on your single server, how many databases do you have ?
I mean, each user should point on one particular database on your server ?

**Comment by mdonogma** [comment]
It’s based on login credentials.

If they login to tenant abc it will sync with tenant abc.

In SaaS solutions dbs are typically split by tenant name.

There can be n-number of tenant names.

So essentially when user logs in, tries to sync it sets the fbcontext for that sync session. If multiple users login and sync context is still based on the tenant the logged into.


A perfect example of how this is done is in

https://aspnetboilerplate.com/

Sent from my iPhone

On 18 Feb 2020, at 04:49, Sébastien Pertus <notifications@github.com> wrote:

﻿

on your single server, how many databases do you have ?
I mean, each user should point on one particular database on your server ?

**Comment by Mimetis** [comment]
This issue has been fixed with the release of the version `v0.5`.
If you still experiencing the same bug, please comment here, I will re-open the issue.
